ensight_lib/include/ensightasciiwriter.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTASCIIWRITER_H
+#define ENSIGHTASCIIWRITER_H
+
+#include <string>
+
+class EnsightObj;
+class QString;
+
+/**
+ * @brief The EnsightAsciiWriter class provides methods to write an Ensight case file with corresponding geometry and variable files
+ * in Ensight Gold Ascii format.
+ */
+class EnsightAsciiWriter
```

Comment:
Why is the design not symmetric with the *Reader classes? That is, in the Readers the Ascii/BinaryReaders have their read*() functions in a detail namespace implying that the user should not use them directly, while the *Writers seem to be usable either directly or via the general EnsightWriter.

Reply:
This must have been an oversight (i.e. incomplete refactoring).
I changed it to be symmetrical, so it reflects that these too should not be used directly.